src/cljc/harja/pvm.cljc

```diff
@@ -691,6 +691,25 @@ kello 00:00:00.000 ja loppu on kuukauden viimeinen päivä kello 23:59:59.999 ."
                          (range (inc ensimmainen-vuosi) viimeinen-vuosi))
                    [[(vuoden-eka-pvm viimeinen-vuosi) loppupvm]])))))
 
+(defn urakan-vuosikolmannekset [alkupvm loppupvm]
+  (let [kolmannesten-kuukaudet [[1 4] [5 8] [9 12]]
+        vuodet (range (vuosi alkupvm) (inc (vuosi loppupvm)))
+        vuosikolmannekset (fn [vuosi kolmannes [alku-kk loppu-kk]]
+                            {kolmannes {:alku (t/first-day-of-the-month (t/date-time vuosi alku-kk))
+                                        :loppu (t/last-day-of-the-month (t/date-time vuosi loppu-kk))}})]
+    (reduce
+      (fn [edellinen vuosi]
+        (merge
+          edellinen
+          (let [vuosikolmannekset
+                (apply
+                  merge
+                  (map-indexed
+                    (fn [i kuukaudet] (vuosikolmannekset vuosi (inc i) kuukaudet))
+                    kolmannesten-kuukaudet))]
+            {vuosi vuosikolmannekset})))
+      {}
+      vuodet)))
 
```

Comment:
Kokeilin tiristää vielä rivejä pois ja sain itse tuollaisen:
```(defn urakan-vuosikolmannekset [alkupvm loppupvm]
  (let [kk [[1 4] [5 8] [9 12]]
        kolmannes (fn [v alku-kk loppu-kk]
                    {:alku (t/first-day-of-the-month v alku-kk)
                     :loppu (t/last-day-of-the-month v loppu-kk)})]
    (into {}
          (map (juxt identity (fn [vuosi]
                                {1 (kolmannes vuosi 1 4)
                                 2 (kolmannes vuosi 5 8)
                                 3 (kolmannes vuosi 9 12)})))
          (range (vuosi alkupvm) (inc (vuosi loppupvm))))))
```

En väitä, että tuo on välttämättä parempi, mutta tuo `merge` käyttö näytti paikalta, jossa voisi `into` käyttää.